### PR TITLE
Fix pagination issue

### PR DESCRIPTION
**Pull request by connorpgpmcelroy**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-8632
https://tools.hmcts.net/jira/browse/EUI-8549

### Change description ###
Fix pagination issue
Also ensure filter resets lists correctly for each task list

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: src/work-allocation/containers/all-work-task/all-work-task.component.ts [hunk 1]
```diff
@@ -46,6 +46,8 @@ export class AllWorkTaskComponent extends TaskListWrapperComponent {
   private selectedTaskType: string = 'All';
   private selectedTaskName: string = '';
 
+  private initialFilterRan = false;
```
Comment (DavidJayakumar) [inline src/work-allocation/containers/all-work-task/all-work-task.component.ts:49]: Name can be isInitialFilterRange and type is missing

File: src/work-allocation/containers/task-list-wrapper/task-list-wrapper.component.ts [hunk 2]
```diff
@@ -55,8 +55,8 @@ export class TaskListWrapperComponent implements OnDestroy, OnInit {
   public isUpdatedTaskPermissions$: Observable<boolean>;
   public updatedTaskPermission: boolean;
   public userRoleCategory: string;
-  private goneBackOne = false;
   private initialFilterApplied = false;
+  private goneBackCount = 0;
```
Comment (DavidJayakumar) [inline src/work-allocation/containers/task-list-wrapper/task-list-wrapper.component.ts:59]: Specify type

**Review by DavidJayakumar — CHANGES_REQUESTED** [review]
@connorpgpmcelroy Please see my comments

**Review by udaydenduluri33 — DISMISSED** [review]
LGTM

**Review by DavidJayakumar — APPROVED** [review]
LGTM

**Review by Mo-Lala84 — APPROVED** [review]
approved

**Review by Mo-Lala84 — APPROVED** [review]
approved

**Review by udaydenduluri33 — APPROVED** [review]

**Pull request merged**